contrib/cirrus/setup_environment.sh

```diff
@@ -227,7 +227,7 @@ case "$TEST_FLAVOR" in
         ;;
     docker-py)
         remove_packaged_podman_files
-        make && make install PREFIX=/usr ETCDIR=/etc
+        make; make install PREFIX=/usr ETCDIR=/etc
```

Comment:
I don't understand the reason for this. My best guess is, you're guessing that the `&&` hides an error? It shouldn't, as verified by:
```console
$ false && true
$ echo $?
1
```
I still lean toward: if git broke our workflow, let's just silence the new git behavior. I want the least possible amount of fiddling with our sensitive code paths.

Reply:
There are two issues here.  1. Handling the new git checks and 2. something is failing and not passing an exit code.

1. Yes, I considered that,  but there are multiple contexts here, root and rootless.  Since we already chown all the files, I thought simply moving that operation would be safer than setting a system-wide git config declaring all of $GOPATH "safe".
2. The `s/&/;/` was a complete shot in the dark while I waited for you to look.  I don't understand the problem, clearly [make is barking about a real exit code](exit status 128) but somehow that isn't getting passed along.  Is the problem in the `Makefile` somehow?

Reply:
clarification: adding a `git config --system --add safe.directory $GOSRC` is what makes me nervous.  Mainly because I don't know/understand all the implications.

Reply:
Yeah, I follow you on risk, and clearly "risk" is relative to our own set of "unknowns" :smile:  The file-ownership issue is something I'm very familiar with.  We don't want to end up in a place later (irrespective of git problems) where we're executing 'make' as `$ROOTLESS_USER` but some mix of the files under `$GOPATH` or `$GOSRC` are owned by root.  That's caused all kinds of trouble before, so I'm quite comfortable changing the order-of-ops.

I understand git less well, and the implications of a `git config --system --add safe.directory $GOSRC` (for me) seem more "big picture" risky:

1) How will that affect the golang toolchain and other repos. under `$GOPATH/src/...`?
2) Do I need to mark ***every*** repo under `$GOPATH/src/...` as "safe"?
3) How will this affect users of `hack/get_ci_vm.sh` where `$GOSRC` is a clone of their personal repo.?
4) What other checks are bypassed by marking a directory safe? Do they matter in all automation contexts?

---

The other thing which appears risky to me is about the `Makefile` in general:  Clearly at least one thing is allowed to fail w/o passing an exit code.  What else can fail by this same mechanism?  Automation (very heavily) assumes `make foo` will **always** exit non-zero if something breaks.  The last thing we want automation to do, is provide false-positives when there's a meaningful failure, let alone this specific git issue.

Reply:
Ahh, ha! I bet this false-positive make problem is caused by patterns such as:

```Makefile
GIT_BRANCH ?= $(shell git rev-parse --abbrev-ref HEAD 2>/dev/null)
GIT_BRANCH_CLEAN ?= $(shell echo $(GIT_BRANCH) | sed -e "s/[^[:alnum:]]/-/g")
```

I'm pretty sure that shell isn't spawned with `-e` or `-o pipefail`.  The `git rev-parse` is probably throwing the ownership error, but the pipe-to-sed hides it.  There are at least a few other pipes used inside `$(shell ...)` blobs which do lazy substitution of further `$(shell ...)` values :confounded: But this seems like something than can be easily proven, and if not fixed outright, can be filed away in the "gotcha" list.

Reply:
yep, this is exactly the problem:

```
$ ls
Makefile
$ cat Makefile

GIT_BRANCH ?= $(shell false)
GIT_BRANCH_CLEAN ?= $(shell echo $(GIT_BRANCH) | true)

.PHONY: all
all:
        @echo "The answer is: '$(GIT_BRANCH_CLEAN)'"
        @echo "The End"
$ make; echo $?
The answer is: ''
The End
0
```

Reply:
That doesn't fully explain the `make: *** [Makefile:321: bin/podman] Error 1` I see in the logs, but it looks like you're onto something, so yay.

Reply:
Hmmm, maybe that's caused by make calling make?  In any case, there's some hidden gremlin here that could screw up our own or somebody else's automation.  I'll file an issue to investigate it further and so at least it's a "known" problem someone can refer to in the meanwhile...

Thanks @edsantiago for your help on this...I'm still debating w/n to also do the git config thing or not :thinking:

Reply:
Opened https://github.com/containers/podman/issues/14021